docgen/docgen.go

```diff
@@ -52,18 +53,33 @@ func (p *docPlugin) Generate(req *plugin_go.CodeGeneratorRequest) (*plugin_go.Co
 		}
 	}
 
-	var source *google_protobuf.FileDescriptorProto
+	var source *parser.FileDescWrapper
 	for _, f := range req.GetProtoFile() {
 		if strings.Contains(f.GetName(), "all.proto") {
-			source = f
-			break
+			for _, fileToGenerate := range req.FileToGenerate {
+				if fileToGenerate == f.GetName() {
+					source = &parser.FileDescWrapper{FileDescriptorProto: f}
+					break
+				}
+			}
 		}
 	}
 
 	if source == nil {
 		return nil, fmt.Errorf("no file to generate")
 	}
 
+	source.DependencyMap = make(map[string]*google_protobuf.FileDescriptorProto)
+
+	// generate dependency array for the source
+	for _, d := range source.GetDependency() {
```

Comment:
i'm on the fence about this- on one hand this is probably safer, but on the other, i think we can also expect the `ProtoFile` contents to be accurate (i.e. it contains decriptors for the contents of `FileToGenerate` + their imports), since we handle that in gunk proper. if you want to leave this in, i would also check to make sure that every element in from `GetDependency` was present

Reply:
The reason i opted to loop through the source's dependency list is the requests ProtoFile contained a lot of unneeded files (e.g annotation parsers) in some cases

Reply:
@mgfcmartinez if you still think we can do this better, please send a follow-up PR. I don't want to hold this for much longer.